homeassistant/helpers/template.py

```diff
@@ -1420,6 +1420,7 @@ def __init__(self, hass, limited=False, strict=False):
         self.filters["atan"] = arc_tangent
         self.filters["atan2"] = arc_tangent2
         self.filters["sqrt"] = square_root
+        self.filters["as_datetime"] = dt_util.parse_datetime
```

Comment:
The naming of the filter/method is a concern for me. `as_datetime` is chosen because of `as_timestamp` and `as_local`. Alternatives:

- `parse_timestamp` -> Close to the original source
- `datetime` -> More natural with `| float`, `| int` and other casting the user is already familiar with.

Reply:
As we already have `as_local` or `as_timestamp`, I think `as_datetime` is fine. If we only want use `datetime`, then we need support more conversation as only from a string

Reply:
yeah agreed, it can be confusing with the actual signature of `datetime` in Python. So this might be the better choice.